Source/WebKit/UIProcess/mac/LegacySessionStateCoding.cpp

```diff
@@ -515,20 +516,21 @@ RefPtr<API::Data> encodeLegacySessionState(const SessionState& sessionState)
     CFIndex length = CFDataGetLength(data.get());
 
     size_t bufferSize = length + sizeof(uint32_t);
-    auto buffer = MallocPtr<uint8_t, HistoryEntryDataEncoderMalloc>::tryMalloc(bufferSize);
-    if (!buffer)
+    auto mallocBuffer = MallocPtr<uint8_t, HistoryEntryDataEncoderMalloc>::tryMalloc(bufferSize);
+    if (!mallocBuffer)
         return nullptr;
+    auto buffer = unsafeForgeSpan(mallocBuffer.leakPtr(), bufferSize);
```

Comment:
We should probably make a MallocSpan data type at some point, that's just like MallocPtr but it remembers its size and has a span() accessor.